src/main/scala/org/apache/spark/sql/TiStrategy.scala

```diff
@@ -263,129 +263,72 @@ class TiStrategy(context: SQLContext) extends Strategy with Logging {
     }
   }
 
-  def groupAggregateProjection(
+  private def groupAggregateProjection(
     groupingExpressions: Seq[NamedExpression],
     aggregateExpressions: Seq[AggregateExpression],
     resultExpressions: Seq[NamedExpression],
     projects: Seq[NamedExpression],
     source: TiDBRelation,
     dagReq: TiDAGRequest
   ): Seq[SparkPlan] = {
-    val aliasMap = mutable.HashMap[(Boolean, Expression), Alias]()
-    val avgPushdownRewriteMap = mutable.HashMap[ExprId, List[AggregateExpression]]()
-    val avgFinalRewriteMap = mutable.HashMap[ExprId, List[AggregateExpression]]()
-
-    def newAggregate(aggFunc: AggregateFunction, originalAggExpr: AggregateExpression) =
-      AggregateExpression(
-        aggFunc,
-        originalAggExpr.mode,
-        originalAggExpr.isDistinct,
-        originalAggExpr.resultId
-      )
+    val deterministicAggAliases = aggregateExpressions.collect {
+      case e if e.deterministic => e -> Alias(e.canonicalized, e.toString())()
+    }.toMap
 
-    def newAggregateWithId(aggFunc: AggregateFunction, originalAggExpr: AggregateExpression) =
-      AggregateExpression(
-        aggFunc,
-        originalAggExpr.mode,
-        originalAggExpr.isDistinct,
-        NamedExpression.newExprId
-      )
-
-    def toAlias(expr: AggregateExpression) =
-      if (!expr.deterministic) {
-        Alias(expr, expr.toString())()
-      } else {
-        aliasMap.getOrElseUpdate(
-          (expr.deterministic, expr.canonicalized),
-          Alias(expr, expr.toString)()
-        )
-      }
+    def aliasPushedPartialResult(e: AggregateExpression): Alias = {
+      deterministicAggAliases.getOrElse(e, Alias(e, e.toString())())
+    }
```

Comment:
@Novemser @ilovesoup The only non-deterministic one handled by `TiAggregate` is `First`, which doesn't seem to be necessary to be special-cased here. Any counter examples?

Reply:
Seems true, currently I cannot find any counter example.

Reply:
Do we want to remove this logic? Was trying to add comments to explain why determinism is a concern here and then realized that it's not...

Reply:
There seems to be some issue with `deterministicAggAliases`, originally `aliasMap` was designed to  eliminate duplicated aggregations, e.g. 
SQL:
```sql
select count(col+1),count(1+col) from any_table
```
`count(col + 1)` and `count(1 + col)` will be reduced to only one canonicalized expression like `count(col + 1)`, and push this canonicalized expression down to TiKV, but current implementation doesn't serve the same as the original design(could be verified by executing the above sql).
Similar issue has been discussed here https://github.com/pingcap/tispark/pull/45
We may need some further discussion on this topic. : )

Reply:
Ah, my bad, it should be:

```scala
deterministicAggAliases.getOrElse(e.canonicalized, Alias(e, e.toString())())
```

Reply:
Very pleased to receive your response so fast, BTW I think determinism logic could be simplified as you suggested : )

Reply:
I can open a separate PR for this change. This one is already too big to review and track.